From bbf398e3f133e2c962846838ff63294e3d25d4e7 Mon Sep 17 00:00:00 2001
From: Yogesh Solanke <ysolanke@nvidia.com>
Date: Wed, 13 Jun 2012 11:51:41 +0530
Subject: [PATCH 01/40] SF/G711Dec: Support higher sample rates

Remove hardcoding of sample rate to 8000

Bug 998747

Change-Id: I2425976ef8eca9e0f59e650d9ff2a4ab6de50183
Reviewed-on: http://git-master/r/64747
Reviewed-on: http://git-psac/r/259
Reviewed-by: Lokesh Pathak <lpathak@nvidia.com>
Tested-by: Lokesh Pathak <lpathak@nvidia.com>
---
 include/media/stagefright/OMXCodec.h              |    2 +-
 media/libstagefright/OMXCodec.cpp                 |   10 +++++-----
 media/libstagefright/codecs/g711/dec/SoftG711.cpp |    3 ++-
 media/libstagefright/codecs/g711/dec/SoftG711.h   |    1 +
 4 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/include/media/stagefright/OMXCodec.h b/include/media/stagefright/OMXCodec.h
index 583c3b3..7fa9770 100644
--- a/include/media/stagefright/OMXCodec.h
+++ b/include/media/stagefright/OMXCodec.h
@@ -248,7 +248,7 @@ private:
             int32_t numChannels, int32_t sampleRate, int32_t bitRate,
             int32_t aacProfile, bool isADTS);
 
-    void setG711Format(int32_t numChannels);
+    void setG711Format(int32_t numChannels, int32_t sampleRate);
 
     status_t setVideoPortFormatType(
             OMX_U32 portIndex,
diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index 70de174..88dcb2f 100755
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -527,10 +527,10 @@ status_t OMXCodec::configureCodec(const sp<MetaData> &meta) {
         // These are PCM-like formats with a fixed sample rate but
         // a variable number of channels.
 
-        int32_t numChannels;
+        int32_t numChannels, sampleRate;
         CHECK(meta->findInt32(kKeyChannelCount, &numChannels));
-
-        setG711Format(numChannels);
+        CHECK(meta->findInt32(kKeySampleRate, &sampleRate));
+        setG711Format(numChannels, sampleRate);
     } else if (!strcasecmp(MEDIA_MIMETYPE_AUDIO_RAW, mMIME)) {
         CHECK(!mIsEncoder);
 
@@ -3485,9 +3485,9 @@ status_t OMXCodec::setAACFormat(
     return OK;
 }
 
-void OMXCodec::setG711Format(int32_t numChannels) {
+void OMXCodec::setG711Format(int32_t numChannels, int32_t sampleRate) {
     CHECK(!mIsEncoder);
-    setRawAudioFormat(kPortIndexInput, 8000, numChannels);
+    setRawAudioFormat(kPortIndexInput, sampleRate, numChannels);
 }
 
 void OMXCodec::setImageOutputFormat(
diff --git a/media/libstagefright/codecs/g711/dec/SoftG711.cpp b/media/libstagefright/codecs/g711/dec/SoftG711.cpp
index bcdd3c7..465b1f2 100644
--- a/media/libstagefright/codecs/g711/dec/SoftG711.cpp
+++ b/media/libstagefright/codecs/g711/dec/SoftG711.cpp
@@ -122,7 +122,7 @@ OMX_ERRORTYPE SoftG711::internalGetParameter(
             pcmParams->eChannelMapping[1] = OMX_AUDIO_ChannelRF;
 
             pcmParams->nChannels = mNumChannels;
-            pcmParams->nSamplingRate = 8000;
+            pcmParams->nSamplingRate = mSampleRate;
 
             return OMX_ErrorNone;
         }
@@ -150,6 +150,7 @@ OMX_ERRORTYPE SoftG711::internalSetParameter(
 
             if(pcmParams->nPortIndex == 0) {
                 mNumChannels = pcmParams->nChannels;
+                mSampleRate = pcmParams->nSamplingRate;
             }
 
             return OMX_ErrorNone;
diff --git a/media/libstagefright/codecs/g711/dec/SoftG711.h b/media/libstagefright/codecs/g711/dec/SoftG711.h
index bff0c68..c0498ef 100644
--- a/media/libstagefright/codecs/g711/dec/SoftG711.h
+++ b/media/libstagefright/codecs/g711/dec/SoftG711.h
@@ -47,6 +47,7 @@ private:
 
     bool mIsMLaw;
     OMX_U32 mNumChannels;
+    OMX_U32 mSampleRate;
     bool mSignalledError;
 
     void initPorts();
-- 
1.7.1

